client-src/elements/chromedash-typeahead.ts

```diff
@@ -142,24 +142,20 @@ export class ChromedashTypeahead extends LitElement {
   }
 
   groupCandidates(candidates: Candidate[]): Candidate[] {
-    const groupsSeen = new Set();
-    const groupsSeenTwice = new Set();
-    for (const c of candidates) {
-      if (groupsSeen.has(c.group)) {
-        groupsSeenTwice.add(c.group);
-      } else {
-        groupsSeen.add(c.group);
-      }
-    }
-
-    const groupsSeenTwiceProcessed = new Set();
+    const groupedByGroup = Object.groupBy(candidates, ({ group }) => group);
     const result: Candidate[] = [];
-    for (const c of candidates) {
-      if (!groupsSeenTwice.has(c.group)) {
-        result.push(c);
-      } else if (!groupsSeenTwiceProcessed.has(c.group)) {
-        result.push({group: c.group, name: c.group + ':', doc: c.doc});
-        groupsSeenTwiceProcessed.add(c.group);
+
+    for (const [groupName, groupCandidatesArray] of Object.entries(groupedByGroup)) {
```

Comment:
This might work and would be a nice upgrade that simplifies the code.  However, I think that order is preserved, but that is now implicit so I am not 100% sure.